src/Utils/Helpers.ts

```diff
@@ -88,4 +89,15 @@ const isInComponents = (path: string): boolean => {
 const isInChannels = (path: string): boolean => {
   return path.startsWith('channels.');
 };
-export { compareComponents, isEqual, isInComponents, isInChannels };
+const toJS = (asyncapiYAMLorJSON: any): any => {
+  if (asyncapiYAMLorJSON.constructor && asyncapiYAMLorJSON.constructor.name === 'Object') {
+    return JSON.parse(JSON.stringify(asyncapiYAMLorJSON));
+  } 
```

Comment:
I have the same question. @fmvilas, why are we checking for AsyncAPI document like this:
https://github.com/asyncapi/parser-js/blob/master/lib/utils.js#L108

Reply:
Ok, I understand

```js
asyncapiYAMLorJSON.constructor && asyncapiYAMLorJSON.constructor.name === 'Object'
```

exactly means that you operate on the pure JS object, which isn't any instance of some class, so it can be treated as pure JSON object, so it's very needed here. However, you shouldn't stringify it and parse again, but only return this object so:

```js
  if (asyncapiYAMLorJSON.constructor && asyncapiYAMLorJSON.constructor.name === 'Object') {
    return asyncapiYAMLorJSON;
  } 
```

is enough :)

Reply:
@magicmatatjahu I don't think using 'return asyncapiYAMLorJSON;' is practical here, because I need a copy of this object. the object itself will go through the parser and the parser will change things inside it, like adding extensions.
btw, shouldn't the parser treat the object as immutable? why does it modify the input object? 🤔

Reply:
Ok, I understand. In ParserJS we should create copy and then operate on it, at the moment we operate on original object. Could you create issue for that in ParserJS repo? 

> why does it modify the input object? 🤔

To add needed extensions 😄 

Please also remember that with

```js
return JSON.parse(JSON.stringify(asyncapiYAMLorJSON));
```

you can have problem with circular references between object and `JSON.stringify` doesn't support it -> more info here https://github.com/asyncapi/parser-js/issues/293

So please add comment for this line and describe current problem. We can go with it (we don't have any other solution), but we must remember about it.

Reply:
Done.